scrapper.py

```diff
@@ -1,10 +1,143 @@
+
+    soup = BeautifulSoup(page.content, 'html.parser')
+
+    return soup
+
+def get_number_of_options(soup: str) -> int:
+    """ Function parse scrapped text and return number of options"""
+    
+    text_with_numbers_of_options = soup.find('div',
+                                     class_='a-search-subtitle search-results-nb').text.strip().split()
+    numbers_of_options = int(text_with_numbers_of_options[1]) # number of options
+
+    return numbers_of_options
+
+def get_flat_price(soup: str, prices: list) -> int:
+    """ Function parse scrapped text, add to list of prices new variants and return numbers_of_added_options"""
+    
+    text_with_prices = soup.find('section', class_='a-list a-search-list a-list-with-favs').findAll('div',
+                             class_='a-card__price')
+
+    for each in text_with_prices:
+        text_with_one_flat_price = ''.join(each.text.strip()[:-1].strip().split())
+        prices.append(int(text_with_one_flat_price))
+
+        numbers_of_added_options = len(prices)  # number of options added to list prices
+
+    return numbers_of_added_options 
+
+def get_flat_square(soup: str, squares: list) -> None:
+    """ Function parse scrapped text, add to list of squares new variants and return None"""
+
+    text_with_flats_square = soup.find('section', class_='a-list a-search-list a-list-with-favs').findAll('div',
+                                         class_='a-card__header-left')
+
+    for each in text_with_flats_square:
+        text_with_one_flat_square = each.find('a', class_='a-card__title').text
+        indexes_of_square_number = re.search(r"\d+(\.\d*)?\sм", text_with_one_flat_square) 
+        squares.append(float(text_with_one_flat_square[indexes_of_square_number.start():
+                                                        indexes_of_square_number.end()-1]))
+
+    return None
+
+def get_offer_id(soup: str, id_list: list) -> None:
+    """ Function parse scrapped text, add to list of offer id new variants and return None"""
+
+    text_with_offers_id = soup.find('section', class_='a-list a-search-list a-list-with-favs').findAll('div',
+                             class_='a-card__header-left')
+
+    for each in text_with_offers_id:
+        text_with_one_offer_id = each.find('a', class_='a-card__title').get('href')
+        indexes_of_offer_id = re.search(r"\d+", text_with_one_offer_id)
+        id_list.append(int(text_with_one_offer_id[indexes_of_offer_id.start():
+                                                    indexes_of_offer_id.end()]))
+
+    return None
+
+def get_cost_per_sq_m(id_list: list, squares: list, prices: list) -> dict:
+    """ Function get lists of id, squares and prices and return dictionary format {id : cost per m^2}"""
+        
+    cost_per_sq_m = {id_list[i]: int(prices[i]/squares[i]) for i in range(len(prices))}
+    
+    return cost_per_sq_m
+
+
+
+def main():
+    
```

Comment:
@ProvVlad , could you please add a docstring that explains how to invoke the script with arguments?